lib/galaxy/tools/parser/xml.py

```diff
@@ -195,7 +214,7 @@ def _parse(data_elem, **kwds):
 
             dataset_collector_descriptions = None
             if collection_elem.find( "discover_datasets" ) is not None:
-                dataset_collector_descriptions = dataset_collector_descriptions_from_elem( collection_elem )
+                dataset_collector_descriptions = dataset_collector_descriptions_from_elem( collection_elem, legacy=False )
```

Comment:
Well... the default is never used in this case because a ``discover_datasets`` tag is found.